packages/editor-ui/src/utils/typesUtils.ts

```diff
@@ -77,28 +77,25 @@ export function shorten(s: string, limit: number, keep: number) {
 export const convertPath = (path: string): string => {
 	// TODO: That can for sure be done fancier but for now it works
 	const placeholder = '*___~#^#~___*';
-	let inBrackets = path.match(/\[(.*?)]/g);
-
-	if (inBrackets === null) {
-		inBrackets = [];
-	} else {
-		inBrackets = inBrackets
-			.map((item) => item.slice(1, -1))
-			.map((item) => {
-				if (item.startsWith('"') && item.endsWith('"')) {
-					return item.slice(1, -1);
-				}
-				return item;
-			});
-	}
+	let inBrackets: string[] = path.match(/\[(.*?)]/g) ?? [];
+
+	inBrackets = inBrackets
+		.map((item) => item.slice(1, -1))
+		.map((item) => {
+			if (item.startsWith('"') && item.endsWith('"')) {
+				return item.slice(1, -1);
+			}
+			return item;
+		});
+
 	const withoutBrackets = path.replace(/\[(.*?)]/g, placeholder);
 	const pathParts = withoutBrackets.split('.');
 	const allParts = [] as string[];
 	pathParts.forEach((part) => {
 		let index = part.indexOf(placeholder);
 		while (index !== -1) {
 			if (index === 0) {
-				allParts.push(inBrackets!.shift() as string);
+				allParts.push(inBrackets.shift() as string);
```

Comment:
```suggestion
				allParts.push(inBrackets.shift() ?? '');
```
We could fall back to empty string here to avoid type coercion.

Reply:
Good point, updated.